### PR TITLE
Salesanalyst

### PR DESCRIPTION
**Pull request by beddings81**
Complete iteration 3. Still need to add test for helper methods.

### CODE REVIEW

File: spec/sales_analyst_spec.rb [hunk 1]
```diff
@@ -277,4 +277,37 @@
...
+    )
+    sales_analyst = sales_engine.analyst
+
+    expect(sales_analyst.invoice_paid_in_full?(1)).to be(true)
+    expect(sales_analyst.invoice_paid_in_full?(2)).to be(false)
+    expect(sales_analyst.invoice_paid_in_full?(3)).to be(false)
+  end
+
+  it 'can return the total $ amount of an invoice' do
+     sales_engine = SalesEngine.from_csv(
```
Comment (Pocketzs) [inline spec/sales_analyst_spec.rb]: check line 299 isn't spaced with 3 spaces

File: spec/sales_analyst_spec.rb [hunk 2]
```diff
@@ -277,4 +277,37 @@
...
+
```
Comment (Pocketzs) [inline spec/sales_analyst_spec.rb]: Probably delete line 300

File: spec/sales_analyst_spec.rb [hunk 3]
```diff
@@ -277,4 +277,37 @@
...
   end
+
+  it 'can check if an invoice is paid in full' do
+    sales_engine = SalesEngine.from_csv(
+
+      :items     => './data/test_data/items_test.csv',
+      :merchants => './data/test_data/merchant_invoices_test2.csv',
+      :invoices  => './data/test_data/invoices_transactions_test.csv',
+      :invoice_items => './data/invoice_items.csv',
+      :transactions => './data/test_data/transactions_test.csv',
```
[the remark this answers is not in the document]
Comment (KaraJoHo) [reply spec/sales_analyst_spec.rb:288]: I think we can keep these how they are

**Review by Pocketzs — CHANGES_REQUESTED** [review]

File: spec/sales_analyst_spec.rb (hunk 2 above)
Comment (beddings81) [reply spec/sales_analyst_spec.rb]: will ignore

**Review by Pocketzs — APPROVED** [review]

**Review by KaraJoHo — APPROVED** [review]

**Review by Kledin85 — APPROVED** [review]
looks good

**Pull request merged**